File: UmweltifyProtocol/Models/PartnerAppRequestLogModel.cs
namespace UmweltifyProtocol.Models;

public class PartnerAppRequestLogModel
{
    public string AppId { get; set; } = string.Empty;
    public string Signature { get; set; } = string.Empty;
    public string Nonce { get; set; } = string.Empty;
    
    public string AppUserId { get; set; } = string.Empty;
    public string AppRequestData { get; set; } = string.Empty;
    
    public string SessionId { get; set; } = string.Empty;
}

File: Javascript/src/PartnerAppRequestLogModel.js
class PartnerAppRequestLogModel {
    constructor(appId, signature, nonce, appUserId, appRequestData, sessionId, deviceId="") {
        this.AppId = appId || "";
        this.Signature = signature || "";
        this.Nonce = nonce || "";
        this.AppUserId = appUserId || "";
        this.AppRequestData = appRequestData || "";
        this.SessionId = sessionId || "";
        this.deviceId = deviceId || "";
    }
}

export { PartnerAppRequestLogModel }